cpssEnabler/linuxNoKernelModule/drivers/intDriver.c

```diff
@@ -366,8 +435,24 @@ static int mvIntDrv_release(struct inode *inode, struct file *file)
 	mvIntDrvNumOpened--;
 	if (!mvIntDrvNumOpened) {
 		/* Cleanup */
-		int slot;
+		int i, slot;
 		struct interrupt_slot *sl;
+		struct pci_dev *pdev;
+		u16 control;
+		
+		for (i=0; i<MAX_PCI_DEVS; i++) {
+				pdev = mvintdrv_get_pci_dev_from_ar();
+				if (pdev) {
+						printk("%s: Disabling MSI for devfn %x vendor %x devid %x msi_cap %x msi_enabled %d\n",
+							__func__, pdev->devfn, pdev->vendor, pdev->device, pdev->msi_cap, pdev->msi_enabled);
+
+						pci_read_config_word(pdev, pdev->msi_cap + PCI_MSI_FLAGS, &control);
+						control &= ~PCI_MSI_FLAGS_ENABLE;
+						pci_write_config_word(pdev, pdev->msi_cap + PCI_MSI_FLAGS, control);
+
```

Comment:
Should we be calling pci_disable_msi() here?
Otherwise pdev->msi_enabled will still be set even though disabled in config space

Reply:
Under development, for future version